### PR TITLE
Add map state

### PR DESCRIPTION
**Pull request by movestill**
Add support for map state: https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-map-state.html

Also update CircleCI to 2.0 so tests run again.

### CODE REVIEW

File: docs/StepFunctionDSL.md [hunk 1]
```diff
@@ -295,6 +296,73 @@ Arguments:
...
+```
+version: "1.0"
+timeout: 60
+Pass()
+    """CreateSomeInputs"""
+    result: '$'
+    data:
+        {
+            "the_array": [1, 2, 3, 4, 5],
+            "foo": "bar"
```
Comment (derek-pryor) [inline docs/StepFunctionDSL.md:345]: Missing end bracket

**Review by derek-pryor — APPROVED** [review]
Looks clean and well implemented. I hope it wasn't too hard to figure out the modifications. I hope this cleans up the StepFunctions that already exist.

It would have been nicer to have been able to implement as a control flow statement, but with the number of modifiers I don't think it would have worked well. :/
```py
foreach '$.itempath' in '$.path':
    Pass()
```

**Review by sandyhider — APPROVED** [review]

File: docs/StepFunctionDSL.md (hunk 1 above)
Comment (movestill) [reply docs/StepFunctionDSL.md:345]: Thanks, not sure how I missed notification of your comment.

**Comment by movestill** [comment]
>Looks clean and well implemented. I hope it wasn't too hard to figure out the modifications. I hope this cleans up the StepFunctions that already exist.

I'll admit that it was _significantly_ harder than the last PR I did. It didn't help that I hadn't done step functions w/o heaviside in years, so that made it hard just mapping the AWS spec back to heaviside's DSL.

>It would have been nicer to have been able to implement as a control flow statement, but with the number of modifiers I don't think it would have worked well. :/

Yeah using a `foreach` would be nice. I didn't even consider that!

**Pull request merged**